procedures: native user auth docs (#2187)
codenvy/che-docs: modules/administration-guide/partials/con_gateway.adoc

[id="gateway_{context}"]
= Gateway

> The {prod-short} gateway consists of 3 main parts:
> 
> * *link:https://github.com/traefik/traefik[Traefik]* instance responsible for requests routing
> * *link:https://github.com/oauth2-proxy/oauth2-proxy[OAuth2-proxy]* ensuring authentication and initiating OAuth flow
> * *link:https://github.com/brancz/kube-rbac-proxy[kube-rbac-proxy]* applying {orch-name} Role based access control (RBAC) policies to control access to any {prod-short} resource.

Reviewer: ```suggestion
The {prod-short} gateway has three main roles:

* Routing requests. It uses link:https://github.com/traefik/traefik[Traefik].

* Ensuring authentication with OpenID Connect (OIDC). It uses {link-oauth2-proxy}.

* Applying {orch-name} Role based access control (RBAC) policies to control access to any {prod-short} resource. It uses {link-kube-rbac-proxy}.
```

and define the attrributes in `antora.yml`:
```
    link-oauth2-proxy: link:https://github.com/oauth2-proxy/oauth2-proxy[OAuth2 Proxy]
    link-kube-rbac-proxy: link:https://github.com/brancz/kube-rbac-proxy[kube-rbac-proxy]
```

Reviewer: Disambiguation needed:

This content implies that the gateway is using the same tools on Kubernetes and OpenShift. 

However, the content in proc_configuring-minikube-github-authentication.adoc seems to imply that  OAuth2-proxy is Minikube or Kubernetes only.

Author: on Kubernetes, we use https://github.com/oauth2-proxy/oauth2-proxy and https://github.com/brancz/kube-rbac-proxy. On OpenShift, we use https://github.com/openshift/oauth-proxy/ and https://github.com/openshift/kube-rbac-proxy, which are de-facto same components, only forked and patched to work properly on OpenShift. I'm not sure how to correctly put it here in the docs.

Reviewer: OK, let's put the links in an example file included in the Additional resources sections then.

> 

Reviewer: ```suggestion

{link-oidc} is the default OIDC issuer. The {prod-short} operator deploys it with preconfigured static users. It can serve as a bridge to third party {identity-provider}, such as GitHub.

```

and define the link in `antora.yml`:

```
    link-oidc-issuer: link:https://dexidp.io/[Dex]
```

[id="gateway_{context}"]
= Gateway

The {prod-short} gateway has following roles:

* Routing requests. It uses link:https://github.com/traefik/traefik[Traefik].

* Authenticating users with OpenID Connect (OIDC). It uses {link-oauth2-proxy}.

> * Providing a default OIDC issuer, which can serve as a bridge to third party {identity-provider}, such as GitHub. {link-oidc-issuer} is the default OIDC issuer, preconfigured with static users.
> 

Reviewer: Removing. Minikube only feature. 

```suggestion

```

> {link-oidc} is the default OIDC issuer. The {prod-short} operator deploys it with preconfigured static users. It can serve as a bridge to third party {identity-provider}, such as GitHub.

Reviewer: ```suggestion
{link-oidc-issuer} is the default OIDC issuer. The {prod-short} operator deploys it with preconfigured static users. It can serve as a bridge to third party {identity-provider}, such as GitHub.
```

> The {prod-short} gateway has three main roles:
> 
> * Routing requests. It uses link:https://github.com/traefik/traefik[Traefik].
> 
> * Ensuring authentication with OpenID Connect (OIDC). It uses {link-oauth2-proxy}.
> 
> * Applying {orch-name} Role based access control (RBAC) policies to control access to any {prod-short} resource. It uses {link-kube-rbac-proxy}.
> 
> {link-oidc-issuer} is the default OIDC issuer. The {prod-short} operator deploys it with preconfigured static users. It can serve as a bridge to third party {identity-provider}, such as GitHub.

Reviewer: ```suggestion
The {prod-short} gateway has following roles:

* Routing requests. It uses link:https://github.com/traefik/traefik[Traefik].

* Authenticating users with OpenID Connect (OIDC). It uses {link-oauth2-proxy}.

* Providing a default OIDC issuer, which can serve as a bridge to third party {identity-provider}, such as GitHub. {link-oidc-issuer} is the default OIDC issuer, preconfigured with static users.

* Applying {orch-name} Role based access control (RBAC) policies to control access to any {prod-short} resource. It uses {link-kube-rbac-proxy}. 
```
